obsreport/obsreport_receiver.go

```diff
@@ -143,19 +143,30 @@ func WithLongLivedCtx() StartReceiveOption {
 	}
 }
 
+// Receiver is a helper to add obersvability to a component.Receiver
+type Receiver struct {
+	receiverID config.ComponentID
+	transport  string
+}
+
+func NewReceiver(receiverID config.ComponentID, transport string) *Receiver {
```

Comment:
I'd suggest making this more like the `Exporter` and `Processor` helpers, where there is a `ReceiverSettings` struct that this method accepts rather than discrete parameters.  That can allow for slightly more flexibility down the road in terms of adding additional settings to that struct.

Reply:
Sounds good I've added this in my latest commit